src/main/java/pl/team/carrent/service/EmployeeService.java

```diff
@@ -0,0 +1,89 @@
+package pl.team.carrent.service;
+
+import org.springframework.stereotype.Service;
+import pl.team.carrent.employee.EmployeeNotExistException;
+import pl.team.carrent.model.Employee;
+import pl.team.carrent.repository.EmployeeRepository;
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class EmployeeService {
+
+    private final EmployeeRepository employeeRepository;
+
+    public EmployeeService(EmployeeRepository employeeRepository) {
+        this.employeeRepository = employeeRepository;
+    }
+
+    public Employee addEmployee(Employee employee) {
+        return employeeRepository.save(employee);
+    }
+
+    public void removeEmployee(Employee employee, LocalDate localDate) {
+        employee.setReleaseDate(localDate);
+    }
+
+    public Set<Employee> getAllEmployees() {
+        return new HashSet<>(employeeRepository.findAll());
+    }
+
+    public Employee getEmployeeByUsername(String username) {
+        return employeeRepository.findByUsername(username)
+                .orElseThrow(() -> new EmployeeNotExistException("username: " + username));
+    }
+
+    public Set<Employee> getActiveEmployees() {
+        return employeeRepository
```

Comment:
Można ogarnąć to w repo -> `findByReleaseDateNotNull` czy jakoś tak :)

Reply:
poprawione